### PR TITLE
Cleanup code in new C++ part

### PR DESCRIPTION
**Pull request by rlnx**
:broom: Clean up the code :broom:
- Fix year in copyrights
- Make namespace declarations more C++17-like
- Add `oneapi` namespace 
- Remove `decomposition` namespace
- Fix relative paths in makefiles

### CODE REVIEW

File: cpp/oneapi/dal/backend/interop/data_conversion.hpp [hunk 1]
```diff
@@ -19,8 +19,9 @@
 
 #include "oneapi/dal/common.hpp"
 
-namespace dal::backend::interop {
+namespace oneapi::dal::backend::interop {
 
+// TODO: Remove using namespace
```
Comment (mchernov-intel) [inline cpp/oneapi/dal/backend/interop/data_conversion.hpp:24]: Do not we be able to use 'using namespace'?

**Comment by mchernov-intel** [comment]
waiting for CI...

File: cpp/oneapi/dal/backend/interop/data_conversion.hpp (hunk 1 above)
Comment (rlnx) [reply cpp/oneapi/dal/backend/interop/data_conversion.hpp:24]: In header files NO

**Pull request merged**